app/medInria/medWorkspaceArea.cpp

```diff
@@ -122,7 +102,116 @@ medWorkspaceArea::~medWorkspaceArea(void)
 
 QPixmap medWorkspaceArea::grabScreenshot()
 {
-    return QPixmap::grabWindow(this->currentWorkspace()->stackedViewContainers()->currentWidget()->winId());
+    medAbstractView* currentView = currentWorkspace()->stackedViewContainers()->getFirstSelectedContainerView();
+    if (currentView != nullptr)
+    {
+        return QPixmap::grabWindow(currentView->viewWidget()->winId());
+    }
+    return QPixmap();
+}
+
+void medWorkspaceArea::grabVideo()
+{
+    medAbstractView* view = this->currentWorkspace()->stackedViewContainers()->getFirstSelectedContainerView();
+    if (view)
+    {
+        medAbstractProcess* process = qobject_cast<medAbstractProcess*>(dtkAbstractProcessFactory::instance()->create("msc::ExportVideo"));
+        medAbstractImageView* iview = dynamic_cast<medAbstractImageView*>(view);
+        medTimeLineParameter* timeLine = iview->timeLineParameter();
+
+        // Get number of frames for dialog window
+        int numberOfFrames = 1;
+        if (timeLine)
+        {
+            numberOfFrames = timeLine->numberOfFrame();
+        }
+
+        QVector<int> userParameters = getExportVideoDialogParameters(numberOfFrames);
+
+        if (userParameters.at(0) == 1) // User is ok to run the video export
+        {
+            // Needed to remove the shadow of the dialog window
+            iview->render();
+
+            if (userParameters.at(1) == 0) // Time video (for 4D datasets)
+            {
+                timeLine->unlockTimeLine();
+
+                for (int f=0; f<timeLine->numberOfFrame(); f+=userParameters.at(2))
+                {
+                    timeLine->setFrame(f);
+                    runExportVideoProcess(process);
+                }
+
+                timeLine->lockTimeLine();
+            }
+            else // Rotation video
+            {
+                for (double rotation=0.0; rotation<360.0; rotation+=userParameters.at(2))
+                {
+                    iview->setRotation(rotation);
+                    runExportVideoProcess(process);
+                }
+            }
+
+            // Compute the video and export it
+            process->update();
+        }
+    }
+}
+
+QVector<int> medWorkspaceArea::getExportVideoDialogParameters(int numberOfFrames)
+{
+    mscExportVideoDialog *exportDialog = new mscExportVideoDialog(this, numberOfFrames);
+
+    QVector<int> results;
+
+    if(exportDialog->exec() == QDialog::Accepted)
+    {
+        results.append(1); // Accepted
+
+        QVector<int> resultsFromDialog = exportDialog->value();
+        results.append(resultsFromDialog.at(0));
+        results.append(resultsFromDialog.at(1));
+    }
+    else
+    {
+        results.append(0); // Cancelled
+    }
+    return results;
+}
+
+void medWorkspaceArea::runExportVideoProcess(medAbstractProcess* process)
```

Comment:
i've updated this function. Now, it sends image by image to the process, instead of pixel by pixel (and it's quicker!)